bnd-maven-plugin-parent/bnd-maven-plugin/src/main/java/org/bndtools/maven/BndMavenPlugin.java

```diff
@@ -0,0 +1,165 @@
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
+
+import aQute.bnd.build.Project;
+import aQute.bnd.osgi.Builder;
+import aQute.bnd.osgi.Constants;
+import aQute.bnd.osgi.FileResource;
+import aQute.bnd.osgi.Jar;
+import aQute.bnd.osgi.Resource;
+import aQute.lib.io.IO;
+
+@Mojo(name = "bnd-process", defaultPhase = LifecyclePhase.PROCESS_CLASSES, requiresDependencyResolution = ResolutionScope.COMPILE)
+public class BndMavenPlugin extends AbstractMojo {
+	
+	@Parameter(defaultValue = "${project.build.directory}", readonly = true)
+	private File targetDir;
+	
+	@Parameter(defaultValue = "${project.build.outputDirectory}", readonly = true)
+	private File classesDir;
+	
+	@Parameter(defaultValue = "${project.build.outputDirectory}/META-INF/MANIFEST.MF", readonly = true)
+	private File manifestPath;
+	
+	@Parameter(defaultValue = "${project}", required = true, readonly = true)
+	private MavenProject project;
+	
+	@Parameter(property = "bnd.trace", defaultValue = "false", readonly = true)
+	private boolean trace;
+
+	public void execute() throws MojoExecutionException {
+		File bndFile = new File(project.getBasedir(), Project.BNDFILE);
+		
+		Builder builder = new Builder();
+		builder.setTrace(true);
+		try {
+			if (bndFile.isFile())
+				builder.setProperties(bndFile);
+			
+			// Reject sub-bundle projects
+			Collection<? extends Builder> subs = builder.getSubBuilders();
+			if (subs.size() != 1)
+				throw new MojoExecutionException("Sub-bundles not permitted in a maven build");
+			File builderFile = builder.getPropertiesFile();
+			if (builderFile != null && !bndFile.equals(builderFile))
+				throw new MojoExecutionException("Sub-bundles not permitted in a maven build");
+
+			// Set bnd classpath
+			List<File> classpath = new LinkedList<File>();
+			@SuppressWarnings("unchecked")
+			Set<Artifact> artifacts = project.getArtifacts();
+			for (Artifact artifact : artifacts) {
+				File artifactFile = artifact.getFile();
+				if (artifactFile != null)
+					classpath.add(artifactFile);
+			}
+			classpath.add(classesDir);
+			builder.setClasspath(classpath.toArray(new File[classpath.size()]));
+
+			// Include local project packages automatically
+			String includes = builder.getProperty(Constants.INCLUDERESOURCE);
+			if (includes == null || includes.trim().length() == 0)
+				builder.setProperty(Constants.INCLUDERESOURCE, classesDir.getPath());
```

Comment:
That sounds good. I just thought is was wrong to not add `target/classes` since then it is harder for the user to add something else without also having to themselves specify `target/classes`.

Also what about `target/resources`? I guess there is no way for bnd to process resources in there?